code/game/objects/items/stacks/stack.dm

```diff
@@ -34,21 +40,35 @@
 	return TRUE
 
 /obj/item/stack/Initialize(mapload, new_amount, merge = TRUE)
-	. = ..()
 	if(new_amount != null)
 		amount = new_amount
 	while(amount > max_amount)
 		amount -= max_amount
 		new type(loc, max_amount, FALSE)
 	if(!merge_type)
 		merge_type = type
+	if(material_type)
+		custom_materials = list(getmaterialref(material_type) = (MINERAL_MATERIAL_AMOUNT * amount))
+	. = ..()
 	if(merge)
 		for(var/obj/item/stack/S in loc)
 			if(S.merge_type == merge_type)
 				merge(S)
+	var/list/temp_recipes = get_main_recipes()
+	recipes = temp_recipes.Copy()
+	if(material_type)
+		var/datum/material/M = getmaterialref(material_type) //First/main material
+		for(var/i in M.categories)
+			switch(i)
+				if(MAT_CATEGORY_RIGID)
+					var/list/temp = GLOB.rigid_recipes.Copy()
+					recipes += temp
 	update_weight()
 	update_icon()
 
+/obj/item/stack/proc/get_main_recipes()
+	return list()//empty list
```

Comment:
Add `SHOULD_CALL_PARENT(1)` here because I'm absolutely sure this will be used for a signal at some point